### PR TITLE
add the option to render a custom link component

### PR DESCRIPTION
**Pull request by teleaziz**

### CODE REVIEW

File: packages/react/src/components/Link.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+import { BuilderStoreContext } from '../store/builder-store'
+
+export const Link: React.SFC<React.AnchorHTMLAttributes<
+  HTMLAnchorElement
+>> = props => (
```
Comment (ca136) [inline packages/react/src/components/Link.tsx]: was this formatted by prettier like this? haven't seen wrapping like this before

File: packages/react/src/components/builder-block.component.tsx [hunk 2]
```diff
@@ -450,7 +451,7 @@ export class BuilderBlock extends React.Component<
   getElement(index = 0, state = this.privateState.state): React.ReactNode {
     const { child, fieldName } = this.props
     const block = this.block
-    let TagName = (block.tagName || 'div').toLowerCase()
+    let TagName: string | typeof Link = (block.tagName || 'div').toLowerCase()
```
Comment (ca136) [inline packages/react/src/components/builder-block.component.tsx]: Is it a common pattern to use `typeof` for getting the 'type' of React components?

File: packages/react/src/components/Link.tsx (hunk 1 above)
Comment (teleaziz) [reply packages/react/src/components/Link.tsx]: yes , I ran prettier on both changed files in this PR

File: packages/react/src/components/builder-block.component.tsx (hunk 2 above)
Comment (teleaziz) [reply packages/react/src/components/builder-block.component.tsx]: Not sure, but it's less verbose than writing React.SFC<React.AnchorHTMLAttributes<HTMLAnchorElement>>

**Comment by teleaziz** [comment]
@ca136 I just read your feedback comment above, I'll include an explanation on the link component next pr

**Pull request merged**